switch_mod/project/dispatch.py

```diff
@@ -160,6 +160,22 @@ def init_dispatch_timepoints(m):
     mod.PROJ_DISPATCH_POINTS = Set(
         dimen=2,
         initialize=init_dispatch_timepoints)
+    # make set array showing all timepoints when each project is active
+    # note: it might be cleaner to first define PROJ_ACTIVE_PERIODS, then use that
+    # to define PROJ_ACTIVE_TIMEPOINTS, PROJ_DISPATCH_POINTS, etc.
+    def proj_active_timepoints_rule(m, p):
+        if not hasattr(m, 'proj_active_timepoints_dict'):
+            d = m.proj_active_timepoints_dict = dict()
+            for p, t in m.PROJ_DISPATCH_POINTS:
```

Comment:
Your iterator "p" is overwriting the function's argument of the same name. Rename this to something like "p2" and your weird pop behavior will go away.